### DISCUSSION TITLE
Pinmux v1.5 cfg file generation

### DISCUSSION THREAD

**Discussion by andregirard37** [OE4T/meta-tegra#1262] [Q&A]
Hello,

I am upgrading my Yocto build from Gatesgarth (JetPack 4.4.1) to Dunfell (JetPack 4.6.3). I have custom changes to the pinmux file that I did through the Jetson_Xavier_NX_Pinmux_Configuration_Wescam_v1.06, which seems to be the latest file from a few years ago now. I noticed that after I build, the tegra19x-mb1-pinmux-p3668-a01.cfg file is slightly different in Dunfell, without my custom changes, but affect other pin I didn't touch. The Dunfell generated cdg file has "Pinmux version 1.5" in the title, whereas my original cfg file that I generated has "Pinmux version 1"
I saw in the NVIDIA forum that there was a change related to the PU on FAN_TACH that explains the v1.5 for Jetson AGX Xavier Pinmux. Not sure if that is relevant for my case. I was wondering if there was a way inside meta-tegra to get the python script that would be available from SDK in Linux_for_Tegra, to regenerate the cfg file from the Pinmux Excel sheet v1.06, and that would be a v1.5 version of the tool.
Thanks

**Comment by madisongh** [comment]
We did have [a recent contribution](https://github.com/OE4T/meta-tegra/pull/1242) with a recipe to extract the pinmux-dts2cfg tool from the L4T kit, but it's only gone into `master` and `kirkstone`.  You could try back-porting it if you need to run the conversion tool during the build.  If it's just a one-time update, it might be easier to just unpack the L4T tarball and run it there.

**Reply by madisongh** [reply]
BTW, I'd also recommend using `kirkstone-l4t-r32.7-x` instead of `dunfell`, if you can.
